src/main/java/adris/altoclef/util/control/SlotHandler.java

```diff
@@ -66,11 +67,13 @@ private void clickWindowSlot(int windowSlot, int mouseButton, SlotActionType typ
 
         _mod.getController().clickSlot(syncId, windowSlot, mouseButton, type, player);
     }
-
-    public boolean forceEquipItem(Item toEquip) {
-
+	
+    public boolean forceEquipItem(Item toEquip, boolean IsFood) {
```

Comment:
I would probably change this boolean to a string. The reason being that as the project evolves, there may be other situations or other task chains that need to do something similar to food.

Reply:
The main goal of this boolean is that if it's set to true, it can equip item even if the bot is eating

So... for now i don't see why putting a string instead.

Reply:
I mean, yea that makes sense. I was referring to generalization for the purpose of future expansions to the mod.

The comment here explains the case "if the bot is eating, do not equip item." 

Good practice would be to make this more general "If the bot is <action>, <do/do not> equip the item. This is because this class is meant to handle all slot work across all tasks and task chains.

perhaps instead of IsFood, we could call it **UnInterruptable** so other mod developers know that if a boolean is passed into _force equip item_, it is because they want to make the task take priority over everything else.


A good example for this would be if you want to equip a sword and attack regardless of your current health. This would be the case if you were trapped in a 1x1 hole with a mob. You want to kill it as fast as possible and don't want any other items being put in slot until done. 

I am writing this at 4:30 am though, so maybe I am not fully coherent lol.

Reply:
Okay, i see, i will do another pull after this one to change that boolean name to **UnInterruptable**